Nguyen-Ba-Hung/WPF/WPF_Example/WPF_Example/ViewModels/PersonViewModel.cs

```diff
@@ -0,0 +1,181 @@
+                {
+                    records = value;
+                    OnPropertyChanged(nameof(Records));
+                }
+            }
+        }
+
+        public PersonViewModel()
+        {
+            sheets = new ObservableCollection<DataTable>();
+
+            ImportFile = new RelayCommand<object>(
+                (impF) => true,
+                (impF) => ImportExcel()
+                );
+            ExportFile = new RelayCommand<object>(
+                (expF) => true,
+                (expF) => ExportExcel()
+                );
+            OnKeyEscape = new RelayCommand<object>(KeyEscape);
+        }
+
+        public ICommand ImportFile { get;set;}
+        public ICommand ExportFile { get;set;}
+        //public ICommand MouseHover { get;set;}
+        //public ICommand MouseLeave { get;set;}
+        public ICommand OnKeyEscape { get;set;}
+
+        public void ImportExcel()
+        {
+            if(sheets != null)
+            {
+                sheets.Clear();
+            }
+
+            OpenFileDialog dialog = new OpenFileDialog();
+            dialog.Filter = "Excel Files|*.xls;*.xlsx;*.xlsm";
+
+            if(dialog.ShowDialog() == true)
+            {
+                string filePath = dialog.FileName;
+
+                using(var stream = File.Open(filePath, FileMode.Open, FileAccess.Read))
+                {
+                    using(var reader = ExcelDataReader.ExcelReaderFactory.CreateReader(stream))
+                    {
+                        var config = new ExcelDataSetConfiguration
+                        {
+                            ConfigureDataTable = _ => new ExcelDataTableConfiguration
+                            {
+                                UseHeaderRow = true
+                            }
+                        };
+
+                        var dataSet = reader.AsDataSet(config).Tables;
+
+                        foreach(DataTable data in dataSet)
+                        {
+                            sheets.Add(data);
+                        }
+
+                        Records = sheets.FirstOrDefault();
+                        reader.Close();
+                    }
+                    stream.Close();
+                    MessageBox.Show("Imported");
+                }
+            }
+        }
+
+        public void ExportExcel()
+        {
+            if(sheets.FirstOrDefault() == null)
+            {
+                MessageBox.Show("No data");
+            }
+            else
+            {
+                ExcelPackage.LicenseContext = OfficeOpenXml.LicenseContext.NonCommercial;
+                SaveFileDialog saveDialog = new SaveFileDialog();
+                saveDialog.Filter = "Excel Files|*.xls;*.xlsx;*.xlsm";
+
+                if (saveDialog.ShowDialog() == true)
+                {
+                    using (ExcelPackage package = new ExcelPackage())
+                    {
+                        foreach (DataTable sheet in sheets)
```

Comment:
sửa lại logic xuất dữ liệu ra file Excel bằng cách sử dụng DataTable thay vì build row và column thủ công.
Tham khảo thông tin của EPPlus trong link:
[đọc và xuất dữ liệu](https://github.com/EPPlusSoftware/EPPlus/wiki/Reading-and-Writing-Data)
[load dữ liệu từ DataTable](https://github.com/EPPlusSoftware/EPPlus/wiki/LoadFromDataTable)